subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesKotlinDSLDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,2252 @@
+            }
+        """
+
+        expect:
+        fails 'checkConfiguration'
+        result.assertHasErrorOutput("Could not resolve all files for configuration ':${suiteName}CompileClasspath'.")
+        result.assertHasErrorOutput("Could not find org.apache.commons:commons-lang3:3.11:com.google.guava:guava:30.1.1-jre:.")
+
+        where:
+        suiteDesc           | suiteName   | suiteDeclaration
+        'the default suite' | 'test'      | 'val test by getting(JvmTestSuite::class)'
+        'a custom suite'    | 'integTest' | 'val integTest by registering(JvmTestSuite::class)'
+    }
+
+    def "can NOT add multiple GAV dependencies to #suiteDesc - at the top level (list)"() {
+        given:
+        buildKotlinFile << """
+            plugins {
+                `java-library`
+            }
+
+            ${mavenCentralRepository(GradleDsl.KOTLIN)}
+
+
+            testing {
+                suites {
+                    $suiteDeclaration
+                }
+            }
+
+            dependencies {
+                val ${suiteName}Implementation = configurations.getByName("${suiteName}Implementation")
+                ${suiteName}Implementation(listOf("org.apache.commons:commons-lang3:3.11", "com.google.guava:guava:30.1.1-jre"))
+            }
+        """
+
+        expect:
+        fails 'help'
+        result.assertHasErrorOutput("Cannot convert the provided notation to an object of type Dependency: [org.apache.commons:commons-lang3:3.11, com.google.guava:guava:30.1.1-jre].")
+
+        where:
+        suiteDesc           | suiteName   | suiteDeclaration
+        'the default suite' | 'test'      | 'val test by getting(JvmTestSuite::class)'
+        'a custom suite'    | 'integTest' | 'val integTest by registering(JvmTestSuite::class)'
+    }
+
+    def "can NOT add multiple GAV dependencies to #suiteDesc (list)"() {
+        given:
+        buildKotlinFile << """
+            plugins {
+                `java-library`
+            }
+
+            ${mavenCentralRepository(GradleDsl.KOTLIN)}
+
+            testing {
+                suites {
+                    $suiteDeclaration {
+                        dependencies {
+                            implementation(listOf("org.apache.commons:commons-lang3:3.11", "com.google.guava:guava:30.1.1-jre"))
+                        }
+                    }
+                }
+            }
+
+        """
+
+        expect:
+        fails 'help'
+        result.assertHasErrorOutput("Cannot convert the provided notation to an object of type Dependency: [org.apache.commons:commons-lang3:3.11, com.google.guava:guava:30.1.1-jre].")
+
+        where:
+        suiteDesc           | suiteName   | suiteDeclaration
+        'the default suite' | 'test'      | 'val test by getting(JvmTestSuite::class)'
+        'a custom suite'    | 'integTest' | 'val integTest by registering(JvmTestSuite::class)'
+    }
+
+    def "can NOT add multiple GAV dependencies to #suiteDesc (varargs)"() {
```

Comment:
Likewise, this one will start "working" if we add a named-args overload as planned, and add the dependency instead of not compiling.

Reply:
This is explicitly checking for a script compilation failure.  If we add a named-args overload, then the test will no longer fail to compile, which will fail the test - and indicate to the dev that a previous assumption has changed.  I think this is exactly what we want.  Can you go into more detail why you think this is a problem, I don't think I follow?

Reply:
I'm not sure it's a problem, I just wanted to note that this will become legal.